tests/src/test/java/alluxio/network/LocalFirstPolicyIntegrationTest.java

```diff
@@ -0,0 +1,158 @@
+import alluxio.Process;
+import alluxio.PropertyKey;
+import alluxio.PropertyKey.Template;
+import alluxio.client.WriteType;
+import alluxio.client.file.FileSystem;
+import alluxio.client.file.FileSystemContext;
+import alluxio.client.file.FileSystemTestUtils;
+import alluxio.master.AlluxioMasterProcess;
+import alluxio.master.MasterInquireClient;
+import alluxio.master.MasterProcess;
+import alluxio.master.SingleMasterInquireClient;
+import alluxio.util.network.NetworkAddressUtils;
+import alluxio.worker.AlluxioWorkerProcess;
+import alluxio.worker.WorkerProcess;
+import alluxio.worker.block.BlockWorker;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.powermock.reflect.Whitebox;
+
+import java.util.Map;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+/**
+ * Integration tests for functionality relating to tiered identity.
+ */
+public class LocalFirstPolicyIntegrationTest extends BaseIntegrationTest {
+  private ExecutorService mExecutor;
+
+  @Rule
+  public ConfigurationRule mConf = new ConfigurationRule(conf());
+
+  private static Map<PropertyKey, String> conf() {
+    Map<PropertyKey, String> map = ConfigurationTestUtils.testConfigurationDefaults(
+        NetworkAddressUtils.getLocalHostName(),
+        AlluxioTestDirectory.createTemporaryDirectory("tiered_identity_test").getAbsolutePath());
+    map.put(PropertyKey.MASTER_RPC_PORT, "0");
+    map.put(PropertyKey.MASTER_WEB_PORT, "0");
+    map.put(PropertyKey.WORKER_RPC_PORT, "0");
+    map.put(PropertyKey.WORKER_DATA_PORT, "0");
+    map.put(PropertyKey.WORKER_WEB_PORT, "0");
+
+    // Set rack locality to be strict.
+    map.put(Template.LOCALITY_TIER_WAIT.format(Constants.LOCALITY_RACK), "-1");
+    return map;
+  }
+
+  @Before
+  public void before() {
+    mExecutor = Executors.newCachedThreadPool();
+  }
+
+  @After
+  public void after() {
+    mExecutor.shutdownNow();
+  }
+
+  @Test
+  public void test() throws Exception {
+    MasterProcess master = AlluxioMasterProcess.Factory.create();
+    WorkerProcess worker1 = AlluxioWorkerProcess.Factory
+        .create(TieredIdentityFactory.fromString("node=node1,rack=rack1"));
+    WorkerProcess worker2 = AlluxioWorkerProcess.Factory
+        .create(TieredIdentityFactory.fromString("node=node2,rack=rack2"));
+
+    runProcess(mExecutor, master);
+    runProcess(mExecutor, worker1);
+    runProcess(mExecutor, worker2);
+
+    master.waitForReady();
+    worker1.waitForReady();
+    worker2.waitForReady();
+
+    MasterInquireClient inquireClient = new SingleMasterInquireClient(master.getRpcAddress());
+    FileSystemContext context = FileSystemContext.create(null, inquireClient);
+    FileSystem fs = FileSystem.Factory.get(context);
+
+    // Write to the worker in node1
+    {
+      Whitebox.setInternalState(TieredIdentityFactory.class, "sInstance",
+          TieredIdentityFactory.fromString("node=node1,rack=rack1"));
```

Comment:
Can we avoid the Whitebox usage?

Reply:
this is not easy because we create location policies by reflection and we would need to inject the fake tiered identity into the location policy. We'd probably end up using Powermock to do this anyway, and I view Whitebox > Powermock since it doesn't mess with classloading.

Reply:
if we start using this pattern across multiple files, I think it would make sense to introduce a Resource/Rule to manage the state